example/order/order.go

```diff
@@ -0,0 +1,130 @@
+package order
+
+import (
+	"fmt"
+
+	"github.com/hallgren/eventsourcing"
+)
+
+type Status string
+
+const (
+	Pending  Status = "pending"
+	Complete Status = "complete"
+)
+
+// Aggregate
+// Collects the business rules for the Order
+//
+// Rules
+// 1. Multiple discounts is not allowed
```

Comment:
@lunjon It would be interesting to see if you could implement this rules set. (If you have the time and interest)

Reply:
I'd love to try - I have time this evening

Reply:
How did it go? I'm here if you want to chat about it.

Reply:
I did a contrived sample of it yesterday but I never thought about publishing it. Do you think it would be valuable here somehow?

Reply:
It would be valuable for me to see how others are thinking when they use this module.